src/cascadia/TerminalConnection/AzureConnection.cpp

```diff
@@ -687,6 +687,27 @@ namespace winrt::Microsoft::Terminal::TerminalConnection::implementation
         _state = AzureState::TermConnecting;
     }
 
+    // Method description:
+    // - Helper function to parse the preferred shell type from user settings returned by cloud console API.
+    // We need this function because the field might be missing in the settings
+    // created with old versions of cloud console API.
+    std::optional<utility::string_t> AzureConnection::_ParsePreferredShellType(const web::json::value& settingsResponse)
+    {
+        if (!settingsResponse.has_object_field(L"properties"))
+        {
+            return std::nullopt;
+        }
+
+        const auto userSettings = settingsResponse.at(L"properties");
+        if (!userSettings.has_string_field(L"preferredShellType"))
+        {
+            return std::nullopt;
+        }
+
+        const auto preferredShellTypeValue = userSettings.at(L"preferredShellType");
+        return std::optional(preferredShellTypeValue.as_string());

```

Comment:
If I might suggest a little more telescoping to make this easier to read:

```suggestion
        if (settingsResponse.has_object_field(L"properties"))
        {
	        const auto userSettings = settingsResponse.at(L"properties");
	        if (userSettings.has_string_field(L"preferredShellType"))
	        {
	        	const auto preferredShellTypeValue = userSettings.at(L"preferredShellType");
	            return preferredShellTypeValue.as_string();
	        }
        }
        
        return std::nullopt;
```

You also shouldn't need to construct an optional directly and instead rely on the optional constructor to let you return a string directly.

If you _do_ need to construct the optional directly, prefer `return { blah.as_string() };` :smile: so you don't need to repeat the optional type.

Reply:
sorry, my suggestion text apparently contains tabs (thanks GitHub_)